### PR TITLE
use local file for shielding list

### PR DESCRIPTION
**Pull request by smaeda-ks**
tl;dr;
We're giving up to _manually_ maintain our shielding list HTML in the template.

`shielding.json` is just an output from our `GET /datacenters` API endpoint:
https://developer.fastly.com/reference/api/utils/datacenter/

So going forward, all we need to do is to just update this JSON file.

### Things to consider

1. The file path for `shielding.json` JSON file
2. Error handling
3. How can we ensure `file_get_contents()` is only called when the dialog is actually opened? It seems like the PHP code in this template file is processed when accessing STORES->Configuration (or STORES->Configuration->System).

### CODE REVIEW

**Comment by smaeda-ks** [comment]
@vvuksan @deninchoo @MartinPeverelli Your advice is very much appreciated (especially for point 3 above).

**Comment by smaeda-ks** [comment]
ok, I think I'm seeing it now...

We could add a new API here (e.g., `GetShieldList.php`):
https://github.com/fastly/fastly-magento2/tree/master/Controller/Adminhtml/FastlyCdn/Backend

then define this in the template:
https://github.com/fastly/fastly-magento2/blob/master/view/adminhtml/templates/system/config/dialogs.phtml#L3832-L3835

call it from JS:
https://github.com/fastly/fastly-magento2/blob/518cb618e6a905e6b1ecdb1bbfbf13a8653d2c76/view/adminhtml/web/js/backends.js#L602-L605

and generate options group dynamically using jQuery...

**Comment by Lukas713** [comment]
Hi,

first thing to notice is that you are using `json_decode(file_get_contents(`. You should use interface that magento has provided to you. `\Magento\Framework\Serialize\SerializerInterface::serialize()` and `\Magento\Framework\Filesystem\DriverInterface::fileGetContents()`.

Second thing, you have a good point with API, but why can't you create `GET` request in the moment of dialog opening. When you click `Add` backend, the API can be invoked [here](https://github.com/fastly/fastly-magento2/blob/master/Controller/Adminhtml/FastlyCdn/Backend/CreateBackend.php#L98), before `return` statement. Result will be returned to js function [here](https://github.com/fastly/fastly-magento2/blob/master/view/adminhtml/web/js/backends.js#L416), and <select> field in the dialog will be populated. Same logic can be applied when you want to edit backend. API call will be invoked [here](https://github.com/fastly/fastly-magento2/blob/master/Controller/Adminhtml/FastlyCdn/Backend/GetBackends.php#L80) and the result will contain shield locations that will populate the select.

**Comment by vvuksan** [comment]
@smaeda-ks I am not super keen on having the /datacenters API polled in real-time. It opens some interesting failures if that call fails for whatever reason. I'd prefer we pull the latest version when we tag a new version e.g. in https://github.com/fastly/fastly-magento2/blob/master/update_version.sh.

**Comment by smaeda-ks** [comment]
Right, I'm not planning to implement direct interaction with our public API for this purpose. I will rather use a static file.
